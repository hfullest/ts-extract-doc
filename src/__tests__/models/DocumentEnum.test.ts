import { resolve } from 'path';
import { extractTsToMarkdown, genrateDocument } from '../..';

describe('DocumentClass', () => {
  const sourcePath = resolve(__dirname, '../fixtures/enum.ts');
  it('文档模型解析[toTypeString]:', () => {
    const documents = genrateDocument(sourcePath);
    const result = documents.map((doc) => doc.map((it) => it.toTypeString())?.join('\n\n')).join('\n\n\n');
    expect(result).toMatchSnapshot();
  });

  it('快照', () => {
    const result = extractTsToMarkdown(sourcePath);
    expect(result).toMatchSnapshot();
  });
});
