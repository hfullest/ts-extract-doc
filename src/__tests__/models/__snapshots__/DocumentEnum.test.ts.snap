// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentEnum 【beauty】模版快照 1`] = `
"<h3 id='exampleenum' style='position:relative'>ExampleEnum</h3>

<div>枚举示例\`ExampleEnum\`描述</div>


<h3 id='exampleenum' style='position:relative'>ExampleEnum</h3>

<div>枚举示例\`ExampleEnum\`描述</div>


<h3 id='exampleenum' style='position:relative'>ExampleEnum</h3>

<div>枚举示例\`ExampleEnum\`描述</div>

"
`;

exports[`DocumentEnum 快照 1`] = `
"### ExampleEnum

枚举示例\`ExampleEnum\`描述


### ExampleEnum

枚举示例\`ExampleEnum\`描述

"
`;

exports[`DocumentEnum 文档模型解析[toTypeString]: 1`] = `""`;
